From c5198ad993f4c7a68059f2ccab3bc24ab5220a75 Mon Sep 17 00:00:00 2001
From: "qiang.cheng" <qiang.cheng@amlogic.com>
Date: Thu, 18 May 2023 18:55:05 +0800
Subject: [PATCH 2/6] dts: adjust default sound card device for debian [1/1]

PD#SWPL-121864

Problem:
no audio output in g12b/sm1 debian

Solution:
adjust default sound card device

Verify:
local g12b sm1

Signed-off-by: qiang.cheng <qiang.cheng@amlogic.com>
Change-Id: Ib14e16cf582432589ff29bc10775a71bc4ba7bf1
---
 .../dts/amlogic/g12b_a311d_w400_debian.dts    | 54 ++++++++---------
 .../dts/amlogic/sm1_s905d3_ac200_debian.dts   | 60 +++++++++----------
 2 files changed, 57 insertions(+), 57 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
index c2e72d39..2eefbb1b 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
@@ -531,6 +531,33 @@
 		//aml-audio-card,effect = <&audio_effect>;
 
 		aml-audio-card,dai-link@0 {
+			format = "i2s";
+			mclk-fs = <256>;
+			//continuous-clock;
+			//bitclock-inversion;
+			//frame-inversion;
+			bitclock-master = <&aml_tdmb>;
+			frame-master = <&aml_tdmb>;
+			//bitclock-master = <&tdmbcodec>;
+			//frame-master = <&tdmbcodec>;
+			/* suffix-name, sync with android audio hal
+			 * what's the dai link used for
+			 */
+			suffix-name = "alsaPORT-i2s";
+			cpu {
+				sound-dai = <&aml_tdmb>;
+				dai-tdm-slot-tx-mask = <1 1>;
+				dai-tdm-slot-rx-mask = <1 1>;
+				dai-tdm-slot-num = <2>;
+				dai-tdm-slot-width = <32>;
+				system-clock-frequency = <12288000>;
+			};
+			tdmbcodec: codec {
+				sound-dai = <&amlogic_codec &ad82584f_62>;
+			};
+		};
+
+		aml-audio-card,dai-link@1 {
 			format = "dsp_a";
 			mclk-fs = <512>;
 			//continuous-clock;
@@ -559,33 +586,6 @@
 			};
 		};
 
-		aml-audio-card,dai-link@1 {
-			format = "i2s";
-			mclk-fs = <256>;
-			//continuous-clock;
-			//bitclock-inversion;
-			//frame-inversion;
-			bitclock-master = <&aml_tdmb>;
-			frame-master = <&aml_tdmb>;
-			//bitclock-master = <&tdmbcodec>;
-			//frame-master = <&tdmbcodec>;
-			/* suffix-name, sync with android audio hal
-			 * what's the dai link used for
-			 */
-			suffix-name = "alsaPORT-i2s";
-			cpu {
-				sound-dai = <&aml_tdmb>;
-				dai-tdm-slot-tx-mask = <1 1>;
-				dai-tdm-slot-rx-mask = <1 1>;
-				dai-tdm-slot-num = <2>;
-				dai-tdm-slot-width = <32>;
-				system-clock-frequency = <12288000>;
-			};
-			tdmbcodec: codec {
-				sound-dai = <&amlogic_codec &ad82584f_62>;
-			};
-		};
-
 		aml-audio-card,dai-link@2 {
 			format = "i2s";
 			mclk-fs = <256>;
diff --git a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
index 8e2b9112..94236a59 100644
--- a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
@@ -638,36 +638,6 @@
 		avout_mute-gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
 
 		aml-audio-card,dai-link@0 {
-			format = "dsp_a";
-			mclk-fs = <512>;
-			//continuous-clock;
-			//bitclock-inversion;
-			//frame-inversion;
-			/* master mode */
-			bitclock-master = <&tdma>;
-			frame-master = <&tdma>;
-			/* slave mode */
-			/*
-			 * bitclock-master = <&tdmacodec>;
-			 * frame-master = <&tdmacodec>;
-			 */
-			suffix-name = "alsaPORT-pcm";
-			tdmacpu: cpu {
-				sound-dai = <&tdma>;
-				dai-tdm-slot-tx-mask =
-							<1 1 1 1 1 1 1 1>;
-				dai-tdm-slot-rx-mask =
-							<1 1 1 1 1 1 1 1>;
-				dai-tdm-slot-num = <8>;
-				dai-tdm-slot-width = <32>;
-				system-clock-frequency = <24576000>;
-			};
-			tdmacodec: codec {
-				sound-dai = <&dummy_codec &dummy_codec>;
-			};
-		};
-
-		aml-audio-card,dai-link@1 {
 			format = "i2s";// "dsp_a";
 			mclk-fs = <256>;
 			//continuous-clock;
@@ -704,6 +674,36 @@
 			};
 		};
 
+		aml-audio-card,dai-link@1 {
+			format = "dsp_a";
+			mclk-fs = <512>;
+			//continuous-clock;
+			//bitclock-inversion;
+			//frame-inversion;
+			/* master mode */
+			bitclock-master = <&tdma>;
+			frame-master = <&tdma>;
+			/* slave mode */
+			/*
+			 * bitclock-master = <&tdmacodec>;
+			 * frame-master = <&tdmacodec>;
+			 */
+			suffix-name = "alsaPORT-pcm";
+			tdmacpu: cpu {
+				sound-dai = <&tdma>;
+				dai-tdm-slot-tx-mask =
+							<1 1 1 1 1 1 1 1>;
+				dai-tdm-slot-rx-mask =
+							<1 1 1 1 1 1 1 1>;
+				dai-tdm-slot-num = <8>;
+				dai-tdm-slot-width = <32>;
+				system-clock-frequency = <24576000>;
+			};
+			tdmacodec: codec {
+				sound-dai = <&dummy_codec &dummy_codec>;
+			};
+		};
+
 		aml-audio-card,dai-link@2 {
 			format = "i2s";
 			mclk-fs = <256>;
-- 
2.25.1

